pkg/plugins/backendplugin/instrumentation/instrumentation.go

```diff
@@ -129,31 +130,35 @@ func instrumentPluginRequest(ctx context.Context, cfg Cfg, pluginCtx *backend.Pl
 	return err
 }
 
+func instrumentContext(ctx context.Context, endpoint string, pCtx backend.PluginContext) context.Context {
+	return log.WithContextualAttributes(ctx, []any{"endpoint", endpoint, "dsName", pCtx.DataSourceInstanceSettings.Name, "dsUID", pCtx.DataSourceInstanceSettings.UID, "pluginId", pCtx.PluginID})
```

Comment:
FYI `pCtx.DataSourceInstanceSettings` might be nil.

Reply:
Good shout, thank you! Fixed in fcc2bee17b405d760d726f57fb11b9d186c513cb